RGO.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs

```diff
@@ -5,97 +5,153 @@
-    private readonly EmployeeDocumentService _employeeDocumentService;
-    private readonly Mock<IEmployeeService> _employeeServiceMock;
     private readonly Mock<IUnitOfWork> _unitOfWorkMock;
-
-    private readonly int employeeId = 1;
+    private readonly Mock<IEmployeeService> _employeeServiceMock;
+    private readonly Mock<IEmployeeTypeService> _employeeTypeServiceMock;
+    private readonly EmployeeDocumentService _employeeDocumentService;
 
     public EmployeeDocumentServiceUnitTest()
     {
         _unitOfWorkMock = new Mock<IUnitOfWork>();
         _employeeServiceMock = new Mock<IEmployeeService>();
         _employeeDocumentService = new EmployeeDocumentService(_unitOfWorkMock.Object, _employeeServiceMock.Object);
+        _employeeTypeServiceMock = new Mock<IEmployeeTypeService>();
     }
 
+    int employeeId = 1;
+    static Employee testEmployee = new Employee(EmployeeTestData.EmployeeDto, EmployeeTypeTestData.DeveloperType);
+
     [Fact]
     public async Task SaveEmployeeDocumentPass()
     {
-        var employeeDocDto = new SimpleEmployeeDocumentDto(
-                                                           1,
-                                                           employeeId,
-                                                           "TestFile.pdf",
-                                                           FileCategory.FixedTerm,
-                                                           "TestFileContent",
-                                                           DateTime.Now
-                                                          );
-
+        _employeeTypeServiceMock
+            .Setup(r => r.GetEmployeeType(EmployeeTypeTestData.DeveloperType.Name))
+            .ReturnsAsync(EmployeeTypeTestData.DeveloperType);
+
+        _unitOfWorkMock
+            .Setup(u => u.Employee.Get(It.IsAny<Expression<Func<Employee, bool>>>()))
+            .Returns(
+                new List<Employee>
+                {
+                    new(EmployeeTestData.EmployeeDto, EmployeeTypeTestData.DeveloperType)
+                    {
+                        EmployeeType = new EmployeeType(EmployeeTypeTestData.DeveloperType),
+                        PhysicalAddress = new EmployeeAddress(EmployeeAddressTestData.EmployeeAddressDto),
+                        PostalAddress = new EmployeeAddress(EmployeeAddressTestData.EmployeeAddressDto)
+                    }
+                }.AsQueryable().BuildMock());
+
+        _unitOfWorkMock
+            .Setup(u => u.EmployeeBanking.Get(It.IsAny<Expression<Func<EmployeeBanking, bool>>>()))
+            .Returns(
+                new List<EmployeeBanking>
+                {
+                    new(EmployeeBankingTestData.EmployeeBankingDto)
+                }.AsQueryable().BuildMock());
+
+
+        List<EmployeeRole> empRoles = new List<EmployeeRole> {
+            new EmployeeRole(
+                new EmployeeRoleDto(1, EmployeeTestData.EmployeeDto, new RoleDto(3, "Employee"))
```

Comment:
RoleDto(3, "Employee")

Should we not also add these to the test data?

Reply:
I looked over it, but it would make sense to